cli/src/native.rs

```diff
@@ -80,14 +80,15 @@ pub(crate) async fn run(args: Params) -> Result<()> {
     .await
     .context("Failed to start node")?;
 
-    node.wait_connected_trusted().await?;
-    let mut events = node.event_subscriber();
-
-    // We have nothing else to do, but we want to keep main alive
     while let Ok(ev) = events.recv().await {
         match ev.event {
             // Skip noisy events
             NodeEvent::ShareSamplingResult { .. } => continue,
+            event @ (NodeEvent::FatalDaserError { .. }
+            | NodeEvent::FetchingHeadersFailed { .. }
+            | NodeEvent::NetworkCompromised) => {
```

Comment:
I was thinking if I should add a `is_error` method in `NodeEvent`. What do you think?

Reply:
sure, why not

Reply:
Done. Should we consider a rejected `ShareSamplingResult` an error and display it as a warning?

Reply:
Also should I propagate the `is_error` to JS?

Reply:
I think yes to both

Reply:
done